tests/extract-dfns.js

```diff
@@ -0,0 +1,97 @@
+const { assert } = require('chai');
+const puppeteer = require('puppeteer');
+const path = require('path');
+
+const baseDfn = {
+    id: 'foo',
+    href: 'about:blank#foo',
+    linkingText: [ 'Foo' ],
+    localLinkingText: [],
+    type: 'dfn',
+    for: [],
+    access: 'private',
+    informative: false
+};
+const tests = [
+  {title: "parses a simple <dfn>",
+   html: "<dfn id='foo'>Foo</dfn>",
+   dfns: [{}]
```

Comment:
I thought I'd note that I had some initial difficulty reading this because, without looking at the rest of the code, this suggests that the extraction will produce one empty object. That is, it would be easier to grasp immediately if here and below, this were `[baseDfn]` (or `[Object.assign({}, baseDfn)]`). I realize that means duplicating code, but then code is written once, read many times...